pkg/monitoring/sensors/temperature.go

```diff
@@ -0,0 +1,47 @@
+package sensors
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+)
+
+const readTimeout = time.Second * 10
+const unitCelsius = "centigrade"
+
+type TemperatureSensorInfo struct {
+	SensorName  string  `json:"sensor_name"`
+	Temperature float64 `json:"temperature"`
+	Unit        string  `json:"unit"`
+}
+
+func ReadTemperatureSensors() ([]*TemperatureSensorInfo, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), readTimeout)
+	defer cancel()
+
+	sensors, err := host.SensorsTemperaturesWithContext(ctx)
+	if err != nil {
+		errText := strings.ToLower(err.Error())
+		if strings.Contains(errText, "not implemented") {
+			logrus.Info("[SENSORS] reading temperature sensors is not implemented for this OS")
```

Comment:
With logrus you need to use structure logging, so all context info is indexable for later use. Here `[SENSORS]` is a part of the message, however it should be a tag for the error. Must be like this:
```go
logrus.WithFields(logrus.Fields{
"pkg": "monitoring",
"module": "sensors",
}).Info("reading temperature sensors...")
```

Also, these fields can be set to a variable like this:

```go
moduleLog := logrus.WithFields(logrus.Fields{
"pkg": "monitoring",
"module": "sensors",
})

moduleLog.Info("reading temperature sensors is not implemented for this OS")
moduleLog.Info.WithError(err).Debugf("not supported by BIOS or driver required")
moduleLog.WithError(err).Error("failed to read temperature sensors")
```

Reply:
Yeah, I know about this option. But in all parts of the project, this format is used: "[MODULE_NAME] some message here" (actually, in some modules the square brackets are missing).

I like the idea of introducing the log format that is simpler to parse, but we should agree on it before applying. Should we apply it? If yes, should we replace the old log.XXX() calls with the new format when making changes to file? I think having two different log formats for one file/module is not an option.

@soupdiver ?

Reply:
See DEV-956
I created a ticket for the refactoring

Reply:
I've updated the logging using package-level variable:
```
var logger = log.WithField("package", "sensors")
```
Also the bunch of changes regarding implementation: now it does not report useless values and reports a 'critical' threshold for each sensor (if available)